# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: esteve@fluidinfo.com-20090901121100-1y44w39jkg9d3v34
# target_branch: ../trunk/
# testament_sha1: dcc46beb9e4c09e0f48f59e133a9a6bd381b45fb
# timestamp: 2009-09-01 14:17:13 +0200
# base_revision_id: svn-v4:bbbe8e31-12d6-0310-92fd-\
#   ac37d47ddeeb:trunk:27268
# 
# Begin patch
=== modified file 'twisted/web/_auth/wrapper.py'
--- twisted/web/_auth/wrapper.py	2009-07-26 16:18:18 +0000
+++ twisted/web/_auth/wrapper.py	2009-09-01 10:11:14 +0000
@@ -108,6 +108,7 @@
         """
         authheader = request.getHeader('authorization')
         if not authheader:
+            request.postpath.insert(0, request.prepath.pop())
             return util.DeferredResource(self._login(Anonymous()))
 
         factory, respString = self._selectParseHeader(authheader)

=== modified file 'twisted/web/test/test_httpauth.py'
--- twisted/web/test/test_httpauth.py	2009-07-26 16:18:18 +0000
+++ twisted/web/test/test_httpauth.py	2009-09-01 12:11:00 +0000
@@ -350,7 +350,7 @@
         not have the required I{Authorization} headers.
         """
         request = self.makeRequest([self.childName])
-        child = self.wrapper.getChildWithDefault(self.childName, request)
+        child = getChildForRequest(self.wrapper, request)
         d = request.notifyFinish()
         def cbFinished(result):
             self.assertEquals(request.responseCode, 401)
@@ -558,15 +558,13 @@
         Anonymous requests are allowed if a L{Portal} has an anonymous checker
         registered.
         """
-        unprotectedContents = "contents of the unprotected child resource"
-
-        class UnprotectedResource(Resource):
-
-            def render_GET(self, request):
-                return unprotectedContents
-
-        self.avatars[ANONYMOUS] = UnprotectedResource()
+        unprotectedContents = "contents of the unprotected child resource" 
+
+        self.avatars[ANONYMOUS] = Resource()
+        self.avatars[ANONYMOUS].putChild(
+            self.childName, Data(unprotectedContents, 'text/plain'))
         self.portal.registerChecker(AllowAnonymousAccess())
+
         self.credentialFactories.append(BasicCredentialFactory('example.com'))
         request = self.makeRequest([self.childName])
         child = getChildForRequest(self.wrapper, request)

# Begin bundle
IyBCYXphYXIgcmV2aXNpb24gYnVuZGxlIHY0CiMKQlpoOTFBWSZTWVyfS+MABCBfgARQU+f//38D
miq////wYAcfHloAAD6AAAAAZRTQAGQAxDQ0Gmg0aANMjEHNMTJk0YTBMTTAJgEMEYEYDmmJkyaM
JgmJpgEwCGCMCMA1MhNEBo000Bo0AAADQAAGUo9IDQGgNGhoaaAAAyaPU2oAVJICaAmjFGEyNSn5
U/TCGoZRjU9T1P00UuRvrq4rrd7bUpn+2dz9c38z+z0+jY0rP02nZtS0UmCm67C2N99YCXvTez7b
s11sMVr7raW6kddEYo8v6enZjHCaSlUVVUn7f6Jtx+uKypVcrWXm6IlqMmHPTETRNrNsgoletsZO
8Q8YVCTpThGxC+z4hEhtISCRKaNEkJWlHMgCYYQZAOCs9A/SxCATQh7qTkPSs2jt7rrrOoi67udF
8eNRNUNZfJyhtfuyYUKW/lBS+Er5UoZGbAjXPpVBjs3gTkBKQVQQFUglAMvYTmTVOtqipdk8h7SL
864uqzSlUrG96dVco1+HUta34GVJlx9e67Ui+icHKkuSXYeK5NqyHiS4lJiul7k8nxSZtj12Zd17
JohohujZrhsZKXtbbVSNFTZRWeSSlJKQ11e2JrJaieakvJeppSXL2d+315bOtwSYtCfR5RtlpUYu
p88TLf131k7b2x1dq/bGCnu2t6c9e7s7NrFDbo3ZDHRh+FUptXsHNpOz8Y3t8xH4SSbnJTWwYJNY
Yab1i7C+updhu2dHRxSbEmbV9k1tJ2tkzZuOiT5ue2T+U7zc0dE2Wp2N9r6XEutaTaq+nfozucqT
ySxZ6dMWaTq6dPB0jnHBTYnTjT0Ljbo6mMnBvafxNSzfoq5eaNe5SNVNaXrmpZkTVr2OCTGZ9lOS
nHOym+2Lc+BNbtbNWXFrbmi++VODemxi3tizY4L7mWd2TVhTaqa8cGXLG/IzXsNkl65sfxCypok1
uKTFmW3OOrkpe1mPe2tWSSr1TheQriyYO5Jr17pou53SW2tTBjors02v4Szra06RwX1siuuVizTV
JdhGAw5G5Jqfce/3fQ5qietRzLnzc1ji1FXT2qWv96lUtV09T2x5ty5esZLllzlJ2rEfWb1qVFUh
CIaIZAppkicLJSGJABKzAVFKio15izKLBqlIWLbbf/S0OqckoUw7r+iozq5Hg96npU/3yeb3pKeP
lv6NTecx8vp/LpyjrYrOxvXSbcHy3pLT8n/UfX+0YT7SoYa6n0+POFfRsfNZ+TAO16nc+sn+Ip5E
mLR6n3vzJo+IVJ7G5ZwbX5NEmKlNW7+7U8Gk9fk9Xs/fu9k72/BYi9u5n3ddpaoU3pLOLlxfRPB1
pMEnFMZMVlNTcStUp3r2bhSTwYtyS/U7mC91k8Y69pH+/Y3z38H/F0t2eLo7FyS5zthaujk9aTJz
e90Vtj9DgRnunqU4TAdXnDaySehmcq6qh1Knm/RWKTzWd79CfGGTCqnFLSxVD7Unqk1c8HLP8FTE
cDPV0M09yUln7v+3uin3ylpWPdkkaoKhwanoefhCz2PJi8G3d3PNmwYntfckp6U5p+s9890yJ8SP
dEXJuetOol38Ok9qpFPJFsrn2oqvhPxTdEx8G1eTxI78DqT4Q5QuOHVxGzIjqJ8/25sGbCD3wfKH
F5r5w6yfJOctJ4kqTrJ/4nQZyff6kMoW5+lTwbWhHtI4Rgn/Lm1nFi/MxeUYXq3OcXSTpGtkxFL1
ojxUTgkvKLZxhFJapJSWN0RlnzSMIPk9a6noMeVDP5esWi+yVea/Q7C1Npzc3KTx041DKNcmoZ2x
quwdlr2hTgXI/Vrky1UtItD3PGTR4MMUmzh9q9im97FpwUlzPVrbYucb9ZNkvkwowSWSm3xWtKvU
U5yzmcG3+8/a7j6OmS0cIsT1QWmkmxODmSxGEOixGY42tPlTLXrTqpP6/FyYnLTJnmlTNMblJZrP
K+FTgxinijcwSPr6E3NybFJdr2ljIa0rXC3BJUuqMKk4nwhumKjA/p8Kk3Eea9Js7+3v3cE8Osn+
E90WjtSOxKgyj6J9CdcOkPbqlzuuZw2TiyWSz80uWbDBt7Sz/8XckU4UJBcn0vjA
